es6-shim.js

```diff
@@ -267,7 +267,9 @@
 
   var $String = String;
 
-  var ES = {
+  // Assign these functions to a prototype to give v8 a hint that it
```

Comment:
how does this impact other engines? v8-specific optimizations have hurt other engines' performance in the past, with bluebird for example.